src/speech2text.py

```diff
@@ -285,6 +263,35 @@ def read_input_file_from_array_file(input_file, slurm_array_task_id):
     return new_input_file
 
 
+def convert_to_wav(input_file, tmp_dir):
+    """Pyannote diarization pipeline does handle resampling to ensure 16 kHz and
+    stereo/mono mixing. However, number of supported audio/video formats appears to be 
+    limited and not listed in README. To be sure, we convert all files to .wav beforehand.
+
+    https://huggingface.co/pyannote/speaker-diarization-3.1
+    """
+
+    if str(input_file).lower().endswith(".wav"):
+        logger.info(f".. .. File is already in wav format: {input_file}")
+        return input_file
+
+    if not Path(input_file).is_file():
+        logger.info(f".. .. File does not exist: {input_file}")
+        return None
+
+    converted_file = Path(tmp_dir) / Path(Path(input_file).name).with_suffix(".wav")
+    if Path(converted_file).is_file():
+        logger.info(f".. .. Converted file {converted_file} already exists.")
+        return converted_file
+    try:
+        AudioSegment.from_file(input_file).export(converted_file, format="wav")
+        logger.info(f".. .. File converted to wav: {converted_file}")
+        return converted_file
+    except Exception as err:
+        logger.info(f".. .. Error while converting file: {err}")
+        return None
+
+
```

Comment:
Not sure if we are using the same code for Kubernetes in the future or not, but just checking the suffix of a file is not enough.
How WhisperX and OpenAI Whisper did the format change were calling a subprocess with ffmpeg in the background. Look at [here](https://github.com/m-bain/whisperX/blob/8227807fa9e076901ea4b4fbbf79c9777a6f5e03/whisperx/audio.py#L25).

I suggest we can do the same, but it's not necessary at this point.

Reply:
Yes, the suffix check is hacky and should be replaced with a proper check in the future. ffmpeg in a subprocess works also for the conversion.